### PR TITLE
Version cache is now in base64 + test workflow agent cache support

### PR DESCRIPTION
**Pull request by Pythyu**
### What does this PR do?

  - ``version_cache`` is now a base64 string that is decoded to assure that there's no character escaping error like ``"`` when making the agent cache file.
  - We forgot to add agent cache support to the ``test`` workflow, hence added support.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -21,6 +21,9 @@ on:
       python_runtimes:
         description: 'Included python runtimes in the build'
         required: false
+      version_cache:
+        description: 'Content of the agent-version.cache file'
```
Comment (KSerrania) [inline .github/workflows/test.yaml:25]:
```suggestion
        description: 'Base 64-encoded content of the agent-version.cache file'
```

+ do the same in the `macos.yaml` workflow.

**Review by KSerrania — APPROVED** [review]
Left one comment, otherwise LGTM.

File: .github/workflows/macos.yaml [hunk 2]
```diff
@@ -102,8 +102,9 @@ jobs:
         VERSION: ${{ github.event.inputs.datadog_agent_ref || env.DEFAULT_DATADOG_AGENT_REF }}
       run: |
         bash ./scripts/clone_agent.sh
+        cd $HOME/go/src/github.com/DataDog/datadog-agent
```
Comment (chouetz) [inline .github/workflows/macos.yaml:105]: You might want to change directory before calling `clone_agent.sh` as well. wdyt?

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -79,6 +82,9 @@ jobs:
         VERSION: ${{ github.event.inputs.datadog_agent_ref || env.DEFAULT_DATADOG_AGENT_REF }}
       run: |
         bash ./scripts/clone_agent.sh
+        cd $HOME/go/src/github.com/DataDog/datadog-agent
```
[the remark this answers is not in the document]
Comment (Pythyu) [reply .github/workflows/test.yaml:85]: same as above

**Review by chouetz — APPROVED** [review]
ok

**Pull request merged**